src/main/resources/db/migration/V1__init.sql

```diff
@@ -79,54 +95,75 @@ create table if not exists bookcrossing.t_user (
     primary key (user_id)
 );
 
+comment on table bookcrossing.t_user is 'Basic user data';
+
 create table if not exists bookcrossing.t_user_role (
     user_id integer not null,
     role_id integer not null,
     primary key (user_id, role_id)
 );
 
-alter table bookcrossing.t_action_mail_user
+comment on table bookcrossing.t_user_role is 'Table of user and role accordance';
+
+alter table if exists bookcrossing.t_action_mail_user
     add constraint action_foreign_user
         foreign key (user_id) references bookcrossing.t_user (user_id) ON DELETE CASCADE;
 
-alter table bookcrossing.t_attach
+create index if not exists action_foreign_user on bookcrossing.t_action_mail_user (user_id);
+
+alter table if exists bookcrossing.t_attach
     add constraint attach_foreign_book
         foreign key (attach_id) references bookcrossing.t_book (book_id);
 
-alter table bookcrossing.t_book
+alter table if exists bookcrossing.t_book
     add constraint book_foreign_user
         foreign key (owner_book) references bookcrossing.t_user (user_id);
 
-alter table bookcrossing.t_bookmarks
+create index if not exists book_foreign_user on bookcrossing.t_book (owner_book)
+    where owner_book is not null;
+
+alter table if exists bookcrossing.t_bookmarks
     add constraint bookmarks_foreign_book
         foreign key (book_id) references bookcrossing.t_book (book_id);
 
-alter table bookcrossing.t_bookmarks
+create index if not exists bookmarks_foreign_book on bookcrossing.t_bookmarks (book_id);
+
+alter table if exists bookcrossing.t_bookmarks
     add constraint bookmarks_foreign_user
         foreign key (user_id) references bookcrossing.t_user (user_id);
 
-alter table bookcrossing.t_correspondence
+alter table if exists bookcrossing.t_correspondence
     add constraint correspondence_foreign_first_user
         foreign key (first_user_id) references bookcrossing.t_user (user_id);
 
-alter table bookcrossing.t_correspondence
+alter table if exists bookcrossing.t_correspondence
     add constraint correspondence_foreign_second_user
         foreign key (second_user_id) references bookcrossing.t_user (user_id);
 
-alter table bookcrossing.t_messages
+create index if not exists correspondence_foreign_second_user on bookcrossing.t_correspondence (second_user_id);
+
+alter table if exists bookcrossing.t_messages
     add constraint messages_foreign_correspondence
         foreign key (correspondence_first_user_id, correspondence_second_user_id)
             references bookcrossing.t_correspondence (first_user_id, second_user_id);
 
-alter table bookcrossing.t_messages
+create index if not exists messages_foreign_correspondence on bookcrossing.t_messages (correspondence_first_user_id, correspondence_second_user_id)
+    where (correspondence_first_user_id is not null and correspondence_second_user_id is not null);
```

Comment:
Keep only first part `correspondence_first_user_id is not null` and remove second clause  `and correspondence_second_user_id is not null)`